src/templates/Home/state/index.ts

```diff
@@ -20,106 +20,77 @@ interface HomeActions {
   },
-
   addTodo: (newTodo) => {
-    const result = { ...newTodo, id: uuid() };
     const todos = produce(get().todos, (draft) => {
-      if (!draft[newTodo.columnId]) {
-        draft[newTodo.columnId] = [];
-      }
-
-      draft[newTodo.columnId].push(result);
+      draft.push(newTodo);
     });
-
     set({ todos });
-    return result;
+    return newTodo;
   },
-
-  moveTodo: (newTodo, fromColumnId, toColumnId, position) => {
-    const todos = produce(get().todos, (draftState) => {
-      const sourceColumn = (draftState[fromColumnId] ?? []).filter(
-        (todo) => todo.id !== newTodo.id,
-      );
-
-      let destinationColumn = draftState[toColumnId] ?? [];
-      if (fromColumnId === toColumnId) {
-        destinationColumn = sourceColumn;
-      }
-
-      if (position <= destinationColumn.length) {
-        destinationColumn.splice(position, 0, newTodo);
-      } else {
-        destinationColumn.push(newTodo);
-      }
-
-      draftState[fromColumnId] = sourceColumn;
-      draftState[toColumnId] = destinationColumn;
+  editTodo: (editedTodo) => {
+    const todos = produce(get().todos, (draft) => {
+      const todoIndex = draft.findIndex((todo) => todo.id === editedTodo.id);
+      draft[todoIndex] = {...editedTodo};
     });
-
     set({ todos });
   },
-
-  addColumn: (newColumn) => {
-    const currentState = get();
-    const newId = uuid();
-    const todos = produce(currentState.todos, (draft) => {
-      draft[newId] = [];
-    });
-    const columns = produce(currentState.columns, (draft) => {
-      const position = Object.keys(currentState.columns).length;
-      draft[newId] = { ...newColumn, id: newId, position };
+  moveTodo: (columnFromId, columnToId, todo, newPosition) => {
+    const todos = produce(get().todos, (draft) => {
+      if (columnFromId === columnToId) {
+        // Adjusting positions of todos within the same column
+        const filteredTodos = draft.filter(t => t.columnId === columnFromId && t.id !== todo.id);
+        filteredTodos.forEach(t => {
+          if (t.position > todo.position && t.position <= newPosition) {
+            t.position--;
+          } else if (t.position < todo.position && t.position >= newPosition) {
+            t.position++;
+          }
+        });
+      } else {
+        // Adjusting positions of the todos in the source column
+        const sourceTodos = draft.filter(t => t.columnId === columnFromId && t.position > todo.position);
+        sourceTodos.forEach(t => t.position--);
+  
+        // Adjusting positions of the todos in the destination column
+        const destinationTodos = draft.filter(t => t.columnId === columnToId && t.position >= newPosition);
+        destinationTodos.forEach(t => t.position++);
+      }
+      
+      // Updating the position and column of the moved todo
+      const todoIndex = draft.findIndex(t => t.id === todo.id);
+      draft[todoIndex].columnId = columnToId;
+      draft[todoIndex].position = newPosition;
```

Comment:
The previous code was working. It should be easier to just update was was working rather than write it this way.

Reply:
Fixed